Source/DafnyLanguageServer/Language/Symbols/SymbolTableFactory.cs

```diff
@@ -179,11 +224,17 @@ private class DesignatorVisitor : SyntaxTreeVisitor {
           // Many resolutions for automatically generated nodes (e.g. Decreases, Update when initializating a variable
           // at declaration) cause duplicated visits. These cannot be prevented at this time as it seems there's no way
           // to distinguish nodes from automatically created one (i.e. nodes of the original syntax tree vs. nodes of the
-          // abstract syntax tree). We could just ignore such duplicates. However, we may miss programatic errors if we
-          // do so.
+          // abstract syntax tree). We just ignore such duplicates until more information is availabe in the AST.
           var range = token.GetLspRange();
           SymbolLookup.Add(range.Start, range.End, symbol);
-          designators.Add(node, symbol);
+          if (designators.TryGetValue(node, out var registeredSymbol)) {
+            if (registeredSymbol != symbol) {
+              logger.LogInformation("Conflicting  symbol resolution nf designator named {Identifier} in {Filename}@({Line},{Column})",
```

Comment:
Would you mind adding an example or reference to a unit test that leads to this error? This makes it easier to refactor/test this code in the future.

Reply:
There is currently no unit test that lead to this error so far, since normally there shouldn't. But since we don't have guarantees like Dafny offers, and it happened in the past, we should just keep it.